examples/ola-recorder.cpp

```diff
@@ -115,36 +122,70 @@ int VerifyShow(const string &filename) {
     return ola::EXIT_NOINPUT;
 
   map<unsigned int, unsigned int> frames_by_universe;
-  uint64_t total_time = 0;
 
-  unsigned int universe;
-  ola::DmxBuffer buffer;
-  unsigned int timeout;
+  ShowEntry entry;
   ShowLoader::State state;
+  uint64_t playback_pos = 0;
+  bool playing = false;
   while (true) {
-    state = loader.NextFrame(&universe, &buffer);
+    state = loader.NextEntry(&entry);
     if (state != ShowLoader::OK)
       break;
-    frames_by_universe[universe]++;
-
-    state = loader.NextTimeout(&timeout);
-    if (state != ShowLoader::OK)
+    playback_pos += entry.next_wait;
+    frames_by_universe[entry.universe]++;
+    if (FLAGS_stop > 0 && playback_pos >= FLAGS_stop) {
+      // Compensate for overshooting the stop time
+      playback_pos = FLAGS_stop;
       break;
-    total_time += timeout;
+    }
+    if (!playing && playback_pos > FLAGS_start) {
+      // Found the start point
+      playing = true;
+
+      // Clamp the frames count between 0 and 1
+      map<unsigned int, unsigned int>::iterator clamp_iter;
+      for (clamp_iter = frames_by_universe.begin();
+           clamp_iter != frames_by_universe.end();
+           ++clamp_iter) {
+        if (clamp_iter->second > 1) {
+          clamp_iter->second = 1;
+        }
+      }
```

Comment:
This still didn't smell quite right and I've realised why, see the comment above.